src/plugins/intel_gna/tests/unit/transformations/gna_insert_copy_layer.cpp

```diff
@@ -1519,35 +1486,90 @@ TEST(TransformationTests, InsertCopyLayerNFLvsFLSubgraphTestt) {
-        auto split = ngraph::builder::makeSplit(params, ngraph::element::i64, 1, axis);
+        auto params = std::make_shared<Parameter>(ov::element::i64, input_shape);
+        auto split = ngraph::builder::makeSplit(params, ov::element::i64, 1, axis);
         auto reshape = ov::op::util::reshapeTo(split->output(0), shape);
         auto copy = std::make_shared<ov::intel_gna::op::Copy>(reshape);
-        auto const_value = ngraph::builder::makeConstant(ngraph::element::i64, shape, std::vector<size_t>{1});
-        auto concat = std::make_shared<ngraph::opset8::Concat>(ngraph::OutputVector{copy, const_value}, axis);
+        auto const_value = ngraph::builder::makeConstant(ov::element::i64, shape, std::vector<size_t>{1});
+        auto concat = std::make_shared<ov::opset11::Concat>(ov::OutputVector{copy, const_value}, axis);
 
-        auto result = std::make_shared<ngraph::opset8::Result>(concat);
-        ref_func =
-            std::make_shared<ngraph::Function>(ngraph::ResultVector{result}, ngraph::ParameterVector{params}, "Concat");
+        auto result = std::make_shared<ov::opset11::Result>(concat);
+        ref_func = std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{params}, "Concat");
+    }
+
+    ov::pass::Manager m;
+    m.register_pass<ov::pass::InitNodeInfo>();
+    m.register_pass<ov::intel_gna::pass::InsertCopyBeforeConcatLayer>();
+    m.run_passes(func);
+
+    ASSERT_NO_THROW(check_rt_info(func));
+
+    auto result = compare_functions(func, ref_func);
+    ASSERT_TRUE(result.first);
+}
+
+// [Parameter]              [Parameter]
+//     |                         |
+//  [Slice]                   [Slice]
+//     |           =>            |
+//   [FQ]                      [FQ]
+//     |                         |
+// [NonFunc] [Constant]      [NonFunc]
+//     \      /                 |
+//    [Concat]                [Copy] [Constant]
+//       |                       \      /
+//    [Result]                   [Concat]
+//                                  |
+//                               [Result]
+TEST(TransformationTests, InsertCopyLayerFQConcatTest) {
+    std::shared_ptr<ov::Model> func, ref_func;
+    ov::Shape input_shape{64, 64};
+    ov::Shape sliced_shape{59, 64};
+    std::vector<int64_t> sliced_shape_i64{59, 64};
+    std::vector<float> fqIMin{-2.12}, fqIMax{2.12}, fqOMin{-4.24}, fqOMax{4.24};
+    size_t levels = 65535;
+    size_t axis = 0;
+    auto prc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(InferenceEngine::Precision::FP16);
+
+    {
+        auto params = std::make_shared<Parameter>(ov::element::i64, input_shape);
+        auto slice = ngraph::builder::makeSlice(params, {0, 0}, sliced_shape_i64, {1, 1}, {}, prc);
+        auto fq = ngraph::builder::makeFakeQuantize(slice, prc, levels, {}, fqIMin, fqIMax, fqOMin, fqOMax);
+        auto reshape = ov::op::util::reshapeTo(fq->output(0), sliced_shape);
```

Comment:
Why do you create some operations with ngraph::builder, some with std::make_shared, and the reshape is created from ov::op::util? Would it make any functional difference if everything was done, for example, with std::make_shared? If this can be done, it would be more consistent.

Reply:
ngraph helpers do not include appropriate (int this context) method for reshape creation, so the ov::op:util::reshapeTo is used very widely in the code.